### PR TITLE
[network] refactor network README

### PR DESCRIPTION
**Pull request by gregnazario**
This readme is very outdated, and this should give it some much needed
updates to be more like reality.  In addition, reworded some parts that
make it more straightforward or concise.

I'm looking for feedback into it, and my view on it doesn't have to be the only one.

See it here: https://github.com/gregnazario/libra/blob/readme/network/README.md

### CODE REVIEW

File: network/README.md [hunk 1]
```diff
@@ -71,28 +68,33 @@ running as independent "tasks." The [tokio](https://tokio.rs/) framework is
...
-detected from inbound discovery messages.
+
+* **Peer Manager** &mdash; Listens for incoming connections, and can dial outbound
+connections to other peers.  Demultiplexes and forwards messages to appropriate
+protocol handlers.  Additionally, notifies upstream components of new or closed
+connections.  Optionally can be connected to ConnectivityManager for a network with
+Discovery.
+
+* **Connectivity Manager** &mdash; Ensures connections to known peers found via
+Discovery.  Notifies Peer Manager to make outbound dials, or disconnects based
```
Comment (davidiw) [inline network/README.md:82]: Establishes? This isn't grammatically correct

File: network/README.md [hunk 2]
```diff
@@ -71,28 +68,33 @@ running as independent "tasks." The [tokio](https://tokio.rs/) framework is
...
-if and only if it is an eligible member of the network. Connectivity Manager
-receives addresses of peers from the Discovery component and issues
-dial/disconnect requests to the Peer Manager.
-* **Discovery** &mdash; Uses push-style gossip for discovering new peers and
-updates to addresses of existing peers. On every *tick*, it opens a new
-substream with a randomly selected peer and sends its view of the network to
-this peer. It informs the connectivity manager of any changes to the network
-detected from inbound discovery messages.
+
+* **Peer Manager** &mdash; Listens for incoming connections, and can dial outbound
```
Comment (davidiw) [inline network/README.md:75]: nit on the s/can dial/dials/

**Review by davidiw — APPROVED** [review]
thanks!

File: network/README.md [hunk 3]
```diff
@@ -104,24 +106,23 @@ negotiation, etc.
...
         │   ├── direct_send            # protocol for fire-and-forget style message delivery
-        │   ├── discovery              # protocol for peer discovery and gossip
         │   ├── health_checker         # protocol for health probing
-        │   └── rpc                    # protocol for remote procedure calls
-        ├── sink                       # utilities over message sinks
-        └── validator_network          # network API for consensus and mempool
+        │   ├── network                # components for interaction with applications
+        │   ├── rpc                    # protocol for remote procedure calls
+        │   └── wire                   # protocol for LibraNet handshakes and messaging
+        └── testutils                  # utilities for fuzzing
```
Comment (mimoo) [inline network/README.md:128]: I would say for testing (fuzzing is just part of that)

File: network/README.md [hunk 4]
```diff
@@ -104,24 +106,23 @@ negotiation, etc.
...
-    │       └── transport             # composable transport API
-    ├── noise                         # noise framework for authentication and encryption
+    │       └── transport              # composable transport API
+    ├── network-address                # network addresses and encryption
+    |── simple-onchain-discovery       # protocol for peer discovery
     └── src
-        ├── channel                    # mpsc channel wrapped in IntGauge
         ├── connectivity_manager       # component to ensure connectivity to peers
         ├── interface                  # generic network API
+        ├── noise                      # noise framework for authentication and encryption
```
Comment (mimoo) [inline network/README.md:120]: noise integration* (the noise framework stuff is in crypto/)

File: network/README.md [hunk 5]
```diff
@@ -20,10 +20,7 @@ The network component uses:
 * TCP for reliable transport.
 * [Noise](https://noiseprotocol.org/noise.html) for authentication and full
  end-to-end encryption.
-* [Yamux](https://github.com/hashicorp/yamux/blob/master/spec.md) for
-multiplexing substreams over a single connection.
-* Push-style [gossip](https://en.wikipedia.org/wiki/Gossip_protocol) for peer
-discovery.
+* On-chain addresses for discovery, with a seed peers file for initial discovery.
```
Comment (mimoo) [inline network/README.md:23]: this PR is already improving the README but additional stuff could be link to specs folder, mention that we split network messages per internal components, that we have a component negotiation, etc.

File: network/README.md (hunk 1 above)
Comment (gregnazario) [reply network/README.md:82]: Yeah, Establishes is correct :P

File: network/README.md (hunk 5 above)
Comment (davidiw) [reply network/README.md:23]: That's a good point, this is what @JoshLind did: https://github.com/libra/libra/tree/master/secure/key-manager/

Comment (gregnazario) [reply network/README.md:25]: Spec added

**Pull request closed without merging**